src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2239,14 +2239,14 @@ bool entitiesQuery
     catch (const std::exception &e)
     {
       *err = e.what();
-      LM_E(("Runtime Error (exception in nextSafe(): %s)", e.what()));
+      LM_E(("Runtime Error (exception in nextSafe() at %s:%d: <%s> - query: <%s>)", __FUNCTION__, __LINE__, e.what(), query.toString().c_str()));
       releaseMongoConnection(connection);
       return false;
     }
     catch (...)
     {
       *err = "generic exception at nextSafe()";
-      LM_E(("Runtime Error (generic exception in nextSafe())"));
+      LM_E(("Runtime Error (generic exception in nextSafe() at %s:%d - query: <%s>)", __FUNCTION__, __LINE__, query.toString().c_str()));
```

Comment:
I still don't understand. You should get the exact same func and line in two places.
Are you sure this does exactly what you want?
`__LINE__` will be 2249  and `__FUNCTION__` whatever this very function is called.
And the exact same info you already have in the same log line ...

I'm just trying to see whether we have some misunderstanding here.